src/utils/extra.ts

```diff
@@ -1,34 +1,62 @@
 import { ItemType } from '../constants';
 import {
   DocumentItemExtra,
-  DocumentItemExtraProperties,
   EmbeddedLinkItemExtra,
-  EmbeddedLinkItemExtraProperties,
+  FolderItemExtra,
+  ShortcutItemExtra,
 } from '../interfaces';
 import {
   AppItemExtra,
-  AppItemExtraProperties,
-  FileItemProperties,
+  EtherpadItemExtra,
   LocalFileItemExtra,
   S3FileItemExtra,
 } from '../services';
+import { ImmutableCast } from '@/frontend/types';
 
-export const getFileExtra = (
-  extra?: LocalFileItemExtra,
-): FileItemProperties | undefined => extra?.[ItemType.LOCAL_FILE];
+export const getFileExtra = <
+  U extends LocalFileItemExtra | ImmutableCast<LocalFileItemExtra>,
+>(
+  extra?: U,
+): U[ItemType.LOCAL_FILE] | undefined => extra?.[ItemType.LOCAL_FILE];
```

Comment:
Very cool, I didn't know that `T` and `immutable.Record<T>` would resolve to the same value type when using the same property access pattern (though it makes sense) :sweat_smile:

Reply:
Actually it allows the use of this method in immutable and non-immutable world, with the output being immutable when the input was and vice-versa (not sure it was what you meant 😄 ). This was allowed by the work that we did to be able to access properties of records without the `.get()` method (still a mystery to me to this day).